source/ai.cpp

```diff
@@ -68,7 +68,7 @@ bool IsValidAttackTarget( CChar& mChar, CChar *cTarget )
 				// Young players are not valid targets for NPCs outside of dungeons
 				if( cwmWorldState->ServerData()->YoungPlayerSystem() && mChar.IsNpc() && !cTarget->IsNpc() && IsOnline(( *cTarget )) && cTarget->GetAccount().wFlags.test( AB_FLAGS_YOUNG ) && !cTarget->GetRegion()->IsDungeon() )
 				{
-					if( cTarget->GetSocket() && cTarget->GetTimer( tCHAR_YOUNGMESSAGE ) <= cwmWorldState->GetUICurrentTime() || cwmWorldState->GetOverflow() )
+					if( (cTarget->GetSocket() && cTarget->GetTimer( tCHAR_YOUNGMESSAGE ) <= cwmWorldState->GetUICurrentTime()) || cwmWorldState->GetOverflow() )
```

Comment:
This changes the flow of the logic 🤔 The check should be:
- "if socket is valid, AND if EITHER tCHAR_YOUNGMESSAGE is lower than GetUICurrentTime OR GetOverflow returns true"

but with this change it becomes:
- "if socket is valid AND tCHAR_YOUNGMESSAGE is lower than GetUICurrentTime, OR if GetOverflow returns true"

Reply:
Does commit 9d985de now meet the proper criteria?

Reply:
It does! Thank you :)